airflow/providers/trino/transfers/gcs_to_trino.py

```diff
@@ -0,0 +1,103 @@
+    """
+    Loads a csv file from Google Cloud Storage into a Trino table.
+    Assumptions:
+    1. First row of the csv contains headers
+    2. Trino table with requisite columns is already created
+
+    :param source_bucket: Source GCS bucket that contains the csv
+    :param source_object: csv file including the path
+    :param trino_table: trino table to upload the data
+    :param trino_conn_id: destination trino connection
+    :param gcp_conn_id: (Optional) The connection ID used to connect to Google Cloud and
+        interact with the Google Cloud Storage service.
+    :param delegate_to: The account to impersonate using domain-wide delegation of authority,
+        if any. For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :param impersonation_chain: Optional service account to impersonate using short-term
+        credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account.
+    """
+
+    def __init__(
+        self,
+        *,
+        source_bucket: str,
+        source_object: str,
+        trino_table: str,
+        trino_conn_id: str = "trino_default",
+        gcp_conn_id: str = "google_cloud_default",
+        delegate_to: Optional[str] = None,
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.source_bucket = source_bucket
+        self.source_object = source_object
+        self.trino_table = trino_table
+        self.trino_conn_id = trino_conn_id
+        self.gcp_conn_id = gcp_conn_id
+        self.delegate_to = delegate_to
+        self.impersonation_chain = impersonation_chain
+
+    def execute(self, context: 'Context') -> None:
+        gcs_hook = GCSHook(
+            gcp_conn_id=self.gcp_conn_id,
+            delegate_to=self.delegate_to,
+            impersonation_chain=self.impersonation_chain,
+        )
+
+        trino_hook = TrinoHook(trino_conn_id=self.trino_conn_id)
+
+        with NamedTemporaryFile("w+") as temp_file:
+            self.log.info("Downloading data from %s", self.source_object)
+            gcs_hook.download(
+                bucket_name=self.source_bucket,
+                object_name=self.source_object,
+                filename=temp_file.name,
+            )
+
+            data = list(csv.reader(temp_file))
+            fields = tuple(data[0])
+            rows = []
+            for row in data[1:]:
+                rows.append(tuple(row))
```

Comment:
```suggestion
            data = csv.reader(temp_file)
            fields = tuple(next(data))
            rows = (tuple(row) for row in data)
```

This saves significant memory usage when the downloaded file is large.

Reply:
@uranusjr 
`fields = tuple(next(data))` fails with `TypeError: 'tuple' object is not an iterator` when I run the unit-test.
The line flagged from the test file is `op.execute(None)`

I am not really sure why this fails. The error indicates I am passing in a tuple in place of a csv-reader. But, given the line from the test file at which the failure occurs; I am not sure how I can pass in a csv-reader.

Reply:
> `TypeError: 'tuple' object is not an iterator`

This doesn’t sound right. The `next` call is on a `csv.reader()`, which is not a tuple. Did you mistype this as `next(tuple(data))`?

Reply:
No - I did not mistype that. I was surprised it failed too!

I have pushed the change (I expect CI would fail).